Update documentation for "nf-core modules" commands (#886)
nf-core/tools: README.md

INFO     Successfully synchronised [n] pipelines
```

## Modules

### modules list

> To list all modules available on [nf-core/modules](https://github.com/nf-core/modules), you can use
> `nf-core modules list`, which will print all avilable modules to the terminal.

Reviewer: ```suggestion
To list all modules available on [nf-core/modules](https://github.com/nf-core/modules), you can use
`nf-core modules list`, which will print all available modules to the terminal.
```

```console
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Module Name                    ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ bandage/image                  │
│ bcftools/consensus             │
│ bcftools/filter                │
│ bcftools/isec                  │
│ bcftools/merge                 │
│ bcftools/mpileup               │
│ bcftools/stats                 │
.
.
.
```

### modules install

You can install modules from [nf-core/modules](https://github.com/nf-core/modules) in your pipeline using `nf-core modules install <pipeline_dir>`. A module installed this way will be installed to the `<pipeline_dir>/modules/nf-core/software` directory. Below is an example where we install the `star/align` module.

```console
nf-core modules install .

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 1.13

> ? Tool name: star/align
>              star/align           
>              star/genomegenerate  

Reviewer: ```suggestion
? Tool name: star/align
```

> ? Tool name: star/align
>                   star/align           
>                   star/genomegenerate  

Reviewer: ```suggestion
? Tool name: star/align
             star/align           
             star/genomegenerate  
```

```

### modules create

> When writing a new module, it is best to start from the nf-core module template, which contains help messages and makes it easy to follow the nf-core guidlines for modules. You can create a new module using `nf-core modules create <directory>`, where `<directory>` can be both a clone of nf-core/modules or an nf-core pipeline. The `modules create` command will ask you the relevant questions and create all necessary module files.

Reviewer: ```suggestion
When writing a new module, it is best to start from the nf-core module template, which contains extensive `TODO` messages to make it easier for you to follow nf-core guidelines. You can create a new module using `nf-core modules create <directory>`, where `<directory>` can be either a clone of nf-core/modules or a nf-core pipeline repo. The `nf-core modules create` command will prompt you with the relevant questions in order to create all of necessary module files.
```

> When writing a new module, it is best to start from the nf-core module template, which contains extensive `TODO` messages to make it easier for you to follow nf-core guidelines. You can create a new module using `nf-core modules create <directory>`, where `<directory>` can be either a clone of nf-core/modules or a nf-core pipeline repo. The `nf-core modules create` command will prompt you with the relevant questions in order to create all of necessary module files.

Reviewer: ```suggestion
When writing a new module, it is best to start from the nf-core module template which contains extensive `TODO` messages to make it easier for you to follow nf-core guidelines. You can create a new module using `nf-core modules create <directory>`, where `<directory>` can either be a clone of nf-core/modules or an nf-core pipeline repo. The `nf-core modules create` command will prompt you with the relevant questions in order to create all of the necessary module files.
```

```console
nf-core modules create .

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 1.13

INFO     Press enter to use default values (shown in brackets) or type your own responses. ctrl+click underlined text to open     create.py:75
         links.                                                                                                                               
Name of tool/subtool: mytool/mysubtool
```

### modules create-test-yml

> All modules at [nf-core/modules](https://github.com/nf-core/modules) have to bring their own tests, which are run when parts of the module code changes. To help developers build new modules, the `nf-core modules create-test-yml` command automates much of the work necessary to create test. Specifically, after you have written the nextflow code for your tests in `modules/tests/software/<tool>/<subtool>/main.nf`, this command will run the tests for you and create the `modules/tests/software/<tool>/<subtool>/test.yml` file. For instance, the md5 sums of all files created by running your module will be calculated and written to the `test.yml` file.

Reviewer: ```suggestion
All modules on [nf-core/modules](https://github.com/nf-core/modules) have a strict requirement of being unit tested using minimal test data. To help developers build new modules, the `nf-core modules create-test-yml` command automates the creation of the yaml file required to document the output file `md5sum` and other information generated by the testing. After you have written a minimal Nextflow script to test your module `modules/tests/software/<tool>/<subtool>/main.nf`, this command will run the tests for you and create the `modules/tests/software/<tool>/<subtool>/test.yml` file.
```

```console
nf-core modules create-test-yml         

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

>     nf-core/tools version 1.13.dev0

Reviewer: ```suggestion
    nf-core/tools version 1.13
```

INFO     Press enter to use default values (shown in brackets) or type your own responses                               test_yml_builder.py:51

> ? Tool name: star/align
>                   star/align           
>                   star/genomegenerate

Reviewer: ```suggestion
? Tool name: star/align
```
